### DISCUSSION TITLE
Render OCR Line by Line

### DISCUSSION THREAD

**Discussion by AASHISHAG** [mindee/doctr#1404] [General]
Hi, how can I use doctr to print the OCR of a supermarket receipt line by line?

The supermarket receipt has two columns, one for product name and the other for product price. I want to print the price of a product next to it. Any idea?

Basically, I am looking for OCR output as I see in the image.

`result.render()` messes up when multiple columns are present in the receipt.

**Comment by felixT2K** [comment]
Hi @AASHISHAG 👋🏼 ,

You can use: `json_output = result.export()`

And iter trough the `blocks` and `lines` of a page

doc: https://mindee.github.io/doctr/using_doctr/using_models.html#what-should-i-do-with-the-output

**Comment by AASHISHAG** [comment]
Thank you, @felixT2K, for the response. 

I am using this for loop to browse through the JSON.

```
    # Draw each word
    for block in page["blocks"]:
        for line in block["lines"]:
            for word in line["words"]:
                print(word)
```

However, I am still unable to understand how can I get to know if two words are in the same line?

**Reply by orlandito24** [reply]
Hi @AASHISHAG, I had a similar issue and I ended up having to do it manually. The block-line loop isn't consistent, and won't work for lines that have large spacing between words.

**Reply by AASHISHAG** [reply]
Hi @orlandito24 : Thank you for your response.

What did you mean by manually? Did you write an external processing script?

For me, mapping manually would be impossible, because I want to process 1000s of receipts. :(

**Reply by felixT2K** [reply]
@AASHISHAG you can try to disable the block resolving so everything is "one" block:

```
predictor = ocr_predictor(pretrained=True, resolve_blocks=False)
```

**Reply by felixT2K** [reply]
In this case it should be easier for you to get the line text over the whole doc

**Reply by Afrothundr** [reply]
I was facing a similar issue and `resolve_blocks=False` worked perfectly! cheers 🎉
